plugins/Goals/Visualizations/Goals.php

```diff
@@ -60,6 +60,8 @@ public function beforeLoadDataTable()
 
         if ($this->config->disable_subtable_when_show_goals) {
             $this->config->subtable_controller_action = null;
+            $this->config->show_flatten_table = false;
+            $this->requestConfig->request_parameters_to_modify['flat'] = false;
```

Comment:
This fixes an issue affecting all referrer reports. Conversion metrics are only calculated for the main reports. Subtables don't have any. Therefor `disable_subtable_when_show_goals` is set.
But if you first switch to a flattened view of a normal table report and then switch to the goals view it currently shows the conversion metrics for the flattened report nevertheless (containing only 0 values). With this adjustment it will then automatically switch to the unflattened main report (see new UI tests).